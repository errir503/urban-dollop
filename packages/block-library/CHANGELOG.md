<!-- Learn how to maintain this file at https://github.com/WordPress/gutenberg/tree/HEAD/packages#maintaining-changelogs. -->

## Unreleased

## 3.0.0 (2021-05-14)

### Breaking Changes

-   Drop support for Internet Explorer 11 ([#31110](https://github.com/WordPress/gutenberg/pull/31110)). Learn more at https://make.wordpress.org/core/2021/04/22/ie-11-support-phase-out-plan/.
-   Increase the minimum Node.js version to v12 matching Long Term Support releases ([#31270](https://github.com/WordPress/gutenberg/pull/31270)). Learn more at https://nodejs.org/en/about/releases/.

## 2.29.0 (2021-03-17)

### Bug Fixes

-   Fix a regression where the Cover block migration would not work with a non-default contentPosition ([#29542](https://github.com/WordPress/gutenberg/pull/29542))

## 2.28.0 (2021-02-01)

### New Features

-   Allow setting the `crossOrigin` attribute so the `useTransformImage` hook can use cross-origin sources ([#28255](https://github.com/WordPress/gutenberg/pull/28255/)).

### Bug Fixes

-   Fix a regression where the Cover block migration would not work with custom units for `minHeight` ([#28627](https://github.com/WordPress/gutenberg/pull/28627))

## 2.27.0 (2020-12-17)

### Enhancement

-   File Block: Copy url button is moved to Block toolbar.
-   Code and Preformatted Blocks: delete on backspace from an empty block.

### Bug Fixes

-   Fix a regression where the Cover would not show opacity controls for the default overlay color ([#26625](https://github.com/WordPress/gutenberg/pull/26625)).
-   Fix a regression ([#26545](https://github.com/WordPress/gutenberg/pull/26545)) where the Cover block lost its default background overlay color
    ([#26569](https://github.com/WordPress/gutenberg/pull/26569)).
-   Fix Image Block, reset image dimensions when replace URL. bug mentioned in ([#26333](https://github.com/WordPress/gutenberg/issues/26333)).

### Enhancement

-   File Block: Copy url button is moved to Block toolbar.

## 2.23.0 (2020-09-03)

### Enhancement

-   Site title is now a link.

### New Features

-   Add heading level controls to Site Title block.

## 2.12.0 (2020-01-13)

### Bug Fixes

-   Fixes a regression published in version 2.9.2 that would prevent some build tools from including
    styles provided in the packages build-styles directory.

## 2.7.0 (2019-08-05)

### Enhancements

-   Heading block uses `has-text-align-*` class names rather than inline style for text alignment.
-   Verse block uses `has-text-align-*` class names rather than inline style for text alignment.

### Bug Fixes

-   Fixed insertion of columns in the table block, which now inserts columns for all table sections ([#16410](https://github.com/WordPress/gutenberg/pull/16410))

## 2.6.0 (2019-06-12)

-   Fixed an issue with creating upgraded embed blocks that are not registered ([#15883](https://github.com/WordPress/gutenberg/issues/15883)).

## 2.5.0 (2019-05-21)

-   Add vertical alignment controls to Columns Block ([#13899](https://github.com/WordPress/gutenberg/pull/13899/)).
-   Add vertical alignment controls to Media & Text Block ([#13989](https://github.com/WordPress/gutenberg/pull/13989)).
-   Add `wide` and `full` alignments to Archives block ([#14533](https://github.com/WordPress/gutenberg/pull/14533)).
-   Add `wide` and `full` alignments to Categories block ([#14533](https://github.com/WordPress/gutenberg/pull/14533)).
-   Add all alignment options to RSS block ([#14533](https://github.com/WordPress/gutenberg/pull/14533)).
-   Add all alignment options to Search block ([#14533](https://github.com/WordPress/gutenberg/pull/14533)).
-   Add image fill option and focal point picker to Media & Text block ([#14445](https://github.com/WordPress/gutenberg/pull/14445)).
-   Updated the edit flow of the `image` block, updated the edit icon and unified the image editing in only one UI based on `MediaPlaceholder`

### Bug Fixes

-   fix uncaught error in `columns` block due to accessing a property on an object that might be undefined [#14605](https://github.com/WordPress/gutenberg/pull/14605)

## 2.3.0 (2019-03-06)

### New Feature

-   Add background color controls for the table block.
-   Add new `RSS` block ([#7966](https://github.com/WordPress/gutenberg/pull/7966)).
-   Add new `Search` block ([#13583](https://github.com/WordPress/gutenberg/pull/13583)).

## 2.2.12 (2019-01-03)

## 2.2.11 (2018-12-18)

## 2.2.10 (2018-12-12)

## 2.2.9 (2018-11-30)

## 2.2.8 (2018-11-30)

## 2.2.7 (2018-11-22)

## 2.2.6 (2018-11-21)

## 2.2.5 (2018-11-20)

## 2.2.4 (2018-11-15)

## 2.2.3 (2018-11-12)

### Bug Fixes

-   Add a minimum width for the audio block to fixed floated audio blocks.

## 2.2.2 (2018-11-12)

### Polish

-   Columns Block: Improve usability while editing columns.

## 2.2.1 (2018-11-09)

## 2.2.0 (2018-11-09)

### New Features

-   Gallery Block: Add screen reader support for order of images in gallery.

## 2.1.8 (2018-11-03)

### Polish

-   File Block: Create file blocks when dropping multiple files at once.

## 2.1.7 (2018-10-30)

## 2.1.6 (2018-10-30)

### Bug Fixes

-   Classic Block: Prevent theme styles from italicising the italicise button.
-   Gallery Block: Fix the "Remove Image" button appearing blank when an image is focussed.

## 2.1.5 (2018-10-29)

## 2.1.4 (2018-10-22)

### Bug Fixes

-   Video Block: Set correct media types for the poster image.

## 2.1.3 (2018-10-19)

## 2.1.2 (2018-10-18)

## 2.1.0 (2018-10-10)

### New Features

-   Include the classic block if `wp.oldEditor` is defined.
-   Include the HTML block.

## 2.0.0 (2018-09-05)

### Breaking Change

-   Change how required built-ins are polyfilled with Babel 7 ([#9171](https://github.com/WordPress/gutenberg/pull/9171)). If you're using an environment that has limited or no support for ES2015+ such as lower versions of IE then using [core-js](https://github.com/zloirock/core-js) or [@babel/polyfill](https://babeljs.io/docs/en/next/babel-polyfill) will add support for these methods.

### Deprecations

-   Attribute type coercion has been deprecated. Omit the source to preserve type via serialized comment demarcation.
