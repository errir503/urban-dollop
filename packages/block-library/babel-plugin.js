/**
 * External dependencies
 */
const fs = require( 'fs' );

/**
 * Internal dependencies
 */
const isBlockMetadataExperimental = require( './src/utils/is-block-metadata-experimental' );

/**
 * Creates a babel plugin that replaces experimental block imports with
 * null variable declarations.
 *
 * For example:
 *     import * as experimentalBlock from "./experimental-block";
 * On build becomes:
 *     const experimentalBlock = null;
 *
 * This ensures the dead code elimination removes the experimental blocks modules
 * during the production build.
 *
 * For more context, see https://github.com/WordPress/gutenberg/pull/40655/
 *
 * @param {Function} shouldProcessImport Optional callback to decide whether a given import should be processed.
 * @param {boolean}  isGutenbergPlugin   Whether to run the plugin.
 */
function createBabelPlugin( shouldProcessImport, isGutenbergPlugin ) {
	if ( ! shouldProcessImport ) {
		shouldProcessImport = isImportDeclarationAnExperimentalBlock;
	}
	if ( isGutenbergPlugin === undefined ) {
		// process.env.npm_package_config_IS_GUTENBERG_PLUGIN is a string, not a boolean
		isGutenbergPlugin =
			String( process.env.npm_package_config_IS_GUTENBERG_PLUGIN ) ===
			'true';
	}
	/**
	 * The babel plugin created by createBabelPlugin.
	 *
	 * @see createBabelPlugin.
	 * @param {import('@babel/core')} babel Current Babel object.
	 * @return {import('@babel/core').PluginObj} Babel plugin object.
	 */
	return function babelPlugin( { types: t } ) {
		if ( isGutenbergPlugin ) {
			return {};
		}

		return {
			visitor: {
				ImportDeclaration( path ) {
					// Only process the experimental blocks.
					if ( ! shouldProcessImport( path ) ) {
						return;
					}

					// Get the imported variable name.
					const namespaceSpecifier = path.node.specifiers.find(
						( specifier ) =>
							specifier.type === 'ImportNamespaceSpecifier'
					);
					const { name } = namespaceSpecifier.local;

					path.replaceWith(
						t.variableDeclaration( 'const', [
							t.variableDeclarator(
								t.identifier( name ),
								t.nullLiteral()
							),
						] )
					);
				},
			},
		};
	};
}

/**
 * Tests whether an import declaration refers to an experimental block.
 * In broad strokes, it's a block that says "__experimental" in its block.json file.
 * For details, check the implementation.
 *
 * @param {Object} path Babel.js AST path representing the import declaration,
 * @return {boolean | undefined} Whether the import represents an experimental block.
 */
function isImportDeclarationAnExperimentalBlock( path ) {
	// Only look for wildcard imports like import * as a from "source":
	const { node } = path;
	const namespaceSpecifier = node.specifiers.find(
		( specifier ) => specifier.type === 'ImportNamespaceSpecifier'
	);
	if ( ! namespaceSpecifier || ! namespaceSpecifier.local ) {
		return;
	}

	// Only look for imports starting with ./ and without additional slashes in the path.
	const importedPath = node.source.value;
	if (
		! importedPath ||
		! importedPath.startsWith( './' ) ||
		importedPath.split( '/' ).length > 2
	) {
		return false;
	}

	// Check the imported directory has a related block.json file.
	const blockJsonPath = __dirname + '/src/' + importedPath + '/block.json';
	if ( ! fs.existsSync( blockJsonPath ) ) {
		return false;
	}

	// Read the block.json file related to this block
	const { name } = namespaceSpecifier.local;
	let blockJSONBuffer;
	try {
		blockJSONBuffer = fs.readFileSync( blockJsonPath );
	} catch ( e ) {
		process.stderr.write(
			'Could not read block.json for the module "' +
				importedPath +
				'" imported under name "' +
				name +
				'" from path "' +
				blockJsonPath +
				'"'
		);
		throw e;
	}
	let blockJSON;
	try {
		blockJSON = JSON.parse( blockJSONBuffer );
	} catch ( e ) {
		process.stderr.write(
			'Could not parse block.json for the module "' +
				importedPath +
				'" imported under name "' +
				name +
				'" read from path "' +
				blockJsonPath +
				'"'
		);
		throw e;
	}
	if ( ! isBlockMetadataExperimental( blockJSON ) ) {
		return false;
	}

	return true;
}

const babelPlugin = createBabelPlugin();
babelPlugin.createBabelPlugin = createBabelPlugin;
module.exports = babelPlugin;
