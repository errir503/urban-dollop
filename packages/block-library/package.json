{
	"name": "@wordpress/block-library",
	"version": "3.2.2",
	"description": "Block library for the WordPress editor.",
	"author": "The WordPress Contributors",
	"license": "GPL-2.0-or-later",
	"keywords": [
		"wordpress",
		"gutenberg",
		"blocks"
	],
	"homepage": "https://github.com/WordPress/gutenberg/tree/HEAD/packages/block-library/README.md",
	"repository": {
		"type": "git",
		"url": "https://github.com/WordPress/gutenberg.git",
		"directory": "packages/block-library"
	},
	"bugs": {
		"url": "https://github.com/WordPress/gutenberg/issues"
	},
	"engines": {
		"node": ">=12"
	},
	"main": "build/index.js",
	"module": "build-module/index.js",
	"react-native": "src/index",
	"sideEffects": [
		"build-style/**",
		"src/**/*.scss",
		"src/navigation-link/index.js",
		"src/template-part/index.js",
		"src/query/index.js"
	],
	"dependencies": {
		"@babel/runtime": "^7.13.10",
		"@wordpress/a11y": "file:../a11y",
		"@wordpress/api-fetch": "file:../api-fetch",
		"@wordpress/autop": "file:../autop",
		"@wordpress/blob": "file:../blob",
		"@wordpress/block-editor": "file:../block-editor",
		"@wordpress/blocks": "file:../blocks",
		"@wordpress/components": "file:../components",
		"@wordpress/compose": "file:../compose",
		"@wordpress/core-data": "file:../core-data",
		"@wordpress/data": "file:../data",
		"@wordpress/date": "file:../date",
		"@wordpress/deprecated": "file:../deprecated",
		"@wordpress/dom": "file:../dom",
		"@wordpress/editor": "file:../editor",
		"@wordpress/element": "file:../element",
		"@wordpress/escape-html": "file:../escape-html",
		"@wordpress/hooks": "file:../hooks",
		"@wordpress/i18n": "file:../i18n",
		"@wordpress/icons": "file:../icons",
		"@wordpress/is-shallow-equal": "file:../is-shallow-equal",
		"@wordpress/keycodes": "file:../keycodes",
		"@wordpress/notices": "file:../notices",
		"@wordpress/primitives": "file:../primitives",
		"@wordpress/reusable-blocks": "file:../reusable-blocks",
		"@wordpress/rich-text": "file:../rich-text",
		"@wordpress/server-side-render": "file:../server-side-render",
		"@wordpress/url": "file:../url",
		"@wordpress/viewport": "file:../viewport",
		"classnames": "^2.2.5",
		"fast-average-color": "4.3.0",
		"lodash": "^4.17.21",
		"memize": "^1.1.0",
		"micromodal": "^0.4.6",
		"moment": "^2.22.1",
		"react-easy-crop": "^3.0.0",
		"tinycolor2": "^1.4.2"
	},
	"publishConfig": {
		"access": "public"
	}
}
