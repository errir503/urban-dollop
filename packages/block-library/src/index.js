/**
 * WordPress dependencies
 */
import '@wordpress/core-data';
import '@wordpress/block-editor';
import {
	registerBlockType,
	setDefaultBlockName,
	setFreeformContentHandlerName,
	setUnregisteredTypeHandlerName,
	setGroupingBlockName,
} from '@wordpress/blocks';

/**
 * Internal dependencies
 */
import * as paragraph from './paragraph';
import * as image from './image';
import * as heading from './heading';
import * as quote from './quote';
import * as gallery from './gallery';
import * as archives from './archives';
import * as audio from './audio';
import * as buttons from './buttons';
import * as button from './button';
import * as calendar from './calendar';
import * as categories from './categories';
import * as code from './code';
import * as columns from './columns';
import * as column from './column';
import * as cover from './cover';
import * as embed from './embed';
import * as file from './file';
import * as html from './html';
import * as mediaText from './media-text';
import * as navigation from './navigation';
import * as navigationLink from './navigation-link';
import * as homeLink from './home-link';
import * as latestComments from './latest-comments';
import * as latestPosts from './latest-posts';
import * as legacyWidget from './legacy-widget';
import * as logInOut from './loginout';
import * as list from './list';
import * as missing from './missing';
import * as more from './more';
import * as nextpage from './nextpage';
import * as pageList from './page-list';
import * as preformatted from './preformatted';
import * as pullquote from './pullquote';
import * as reusableBlock from './block';
import * as rss from './rss';
import * as search from './search';
import * as group from './group';
import * as separator from './separator';
import * as shortcode from './shortcode';
import * as spacer from './spacer';
import * as table from './table';
// import * as tableOfContents from './table-of-contents';
import * as textColumns from './text-columns';
import * as verse from './verse';
import * as video from './video';
import * as tagCloud from './tag-cloud';
import * as classic from './freeform';
import * as socialLinks from './social-links';
import * as socialLink from './social-link';

// Full Site Editing Blocks
import * as siteLogo from './site-logo';
import * as siteTagline from './site-tagline';
import * as siteTitle from './site-title';
import * as templatePart from './template-part';
import * as query from './query';
import * as queryLoop from './query-loop';
import * as queryTitle from './query-title';
import * as queryPagination from './query-pagination';
import * as queryPaginationNext from './query-pagination-next';
import * as queryPaginationNumbers from './query-pagination-numbers';
import * as queryPaginationPrevious from './query-pagination-previous';
import * as postNavigationLink from './post-navigation-link';
import * as postTitle from './post-title';
import * as postContent from './post-content';
import * as postAuthor from './post-author';
import * as postComment from './post-comment';
import * as postCommentAuthor from './post-comment-author';
import * as postCommentContent from './post-comment-content';
import * as postCommentDate from './post-comment-date';
import * as postComments from './post-comments';
import * as postCommentsCount from './post-comments-count';
import * as postCommentsForm from './post-comments-form';
import * as postCommentsLink from './post-comments-link';
import * as postDate from './post-date';
import * as postExcerpt from './post-excerpt';
import * as postFeaturedImage from './post-featured-image';
import * as postTerms from './post-terms';
import * as termDescription from './term-description';

/**
 * Function to register an individual block.
 *
 * @param {Object} block The block to be registered.
 *
 */
const registerBlock = ( block ) => {
	if ( ! block ) {
		return;
	}
	const { metadata, settings, name } = block;
	registerBlockType( { name, ...metadata }, settings );
};

/**
 * Function to get all the core blocks in an array.
 *
 * @example
 * ```js
 * import { __experimentalGetCoreBlocks } from '@wordpress/block-library';
 *
 * const coreBlocks = __experimentalGetCoreBlocks();
 * ```
 */
export const __experimentalGetCoreBlocks = () => [
	// Common blocks are grouped at the top to prioritize their display
	// in various contexts â€” like the inserter and auto-complete components.
	paragraph,
	image,
	heading,
	gallery,
	list,
	quote,

	// Register all remaining core blocks.
	shortcode,
	archives,
	audio,
	button,
	buttons,
	calendar,
	categories,
	code,
	columns,
	column,
	cover,
	embed,
	file,
	group,
	window.wp && window.wp.oldEditor ? classic : null, // Only add the classic block in WP Context
	html,
	mediaText,
	latestComments,
	latestPosts,
	legacyWidget,
	missing,
	more,
	nextpage,
	pageList,
	preformatted,
	pullquote,
	rss,
	search,
	separator,
	reusableBlock,
	socialLinks,
	socialLink,
	spacer,
	table,
	// tableOfContents,
	tagCloud,
	textColumns,
	verse,
	video,

	// Theme blocks
	siteLogo,
	siteTagline,
	siteTitle,

	query,
	queryLoop,
	queryTitle,
	queryPagination,
	queryPaginationNext,
	queryPaginationNumbers,
	queryPaginationPrevious,

	postTitle,
	postContent,
	postDate,
	postExcerpt,
	postFeaturedImage,
	postTerms,

	logInOut,
];

/**
 * Function to register core blocks provided by the block editor.
 *
 * @param {Array} blocks An optional array of the core blocks being registered.
 *
 * @example
 * ```js
 * import { registerCoreBlocks } from '@wordpress/block-library';
 *
 * registerCoreBlocks();
 * ```
 */
export const registerCoreBlocks = (
	blocks = __experimentalGetCoreBlocks()
) => {
	blocks.forEach( registerBlock );

	setDefaultBlockName( paragraph.name );
	if ( window.wp && window.wp.oldEditor ) {
		setFreeformContentHandlerName( classic.name );
	}
	setUnregisteredTypeHandlerName( missing.name );
	setGroupingBlockName( group.name );
};

/**
 * Function to register experimental core blocks depending on editor settings.
 *
 * @param {boolean} enableFSEBlocks Whether to enable the full site editing blocks.
 * @example
 * ```js
 * import { __experimentalRegisterExperimentalCoreBlocks } from '@wordpress/block-library';
 *
 * __experimentalRegisterExperimentalCoreBlocks( settings );
 * ```
 */
export const __experimentalRegisterExperimentalCoreBlocks =
	process.env.GUTENBERG_PHASE === 2
		? ( { enableFSEBlocks } = {} ) => {
				[
					navigation,
					navigationLink,
					homeLink,

					// Register Full Site Editing Blocks.
					...( enableFSEBlocks
						? [
								templatePart,
								postAuthor,
								postComment,
								postCommentAuthor,
								postCommentContent,
								postCommentDate,
								postComments,
								postCommentsCount,
								postCommentsForm,
								postCommentsLink,
								postNavigationLink,
								termDescription,
						  ]
						: [] ),
				].forEach( registerBlock );
		  }
		: undefined;
