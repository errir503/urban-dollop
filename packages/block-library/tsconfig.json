{
	"extends": "../../tsconfig.base.json",
	"compilerOptions": {
		"rootDir": "src",
		"declarationDir": "build-types",
		"types": [ "gutenberg-env" ],
		"strictNullChecks": true
	},
	"references": [ { "path": "../element" } ],
	"include": [ "src/**/*.ts", "src/**/*.tsx" ]
}
