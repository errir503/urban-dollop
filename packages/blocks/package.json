{
	"name": "@wordpress/blocks",
	"version": "12.15.0",
	"description": "Block API for WordPress.",
	"author": "The WordPress Contributors",
	"license": "GPL-2.0-or-later",
	"keywords": [
		"wordpress",
		"gutenberg",
		"blocks"
	],
	"homepage": "https://github.com/WordPress/gutenberg/tree/HEAD/packages/blocks/README.md",
	"repository": {
		"type": "git",
		"url": "https://github.com/WordPress/gutenberg.git",
		"directory": "packages/blocks"
	},
	"bugs": {
		"url": "https://github.com/WordPress/gutenberg/issues"
	},
	"engines": {
		"node": ">=12"
	},
	"main": "build/index.js",
	"module": "build-module/index.js",
	"react-native": "src/index",
	"sideEffects": [
		"{src,build,build-module}/{index.js,store/index.js}"
	],
	"dependencies": {
		"@babel/runtime": "^7.16.0",
		"@wordpress/autop": "file:../autop",
		"@wordpress/blob": "file:../blob",
		"@wordpress/block-serialization-default-parser": "file:../block-serialization-default-parser",
		"@wordpress/compose": "file:../compose",
		"@wordpress/data": "file:../data",
		"@wordpress/deprecated": "file:../deprecated",
		"@wordpress/dom": "file:../dom",
		"@wordpress/element": "file:../element",
		"@wordpress/hooks": "file:../hooks",
		"@wordpress/html-entities": "file:../html-entities",
		"@wordpress/i18n": "file:../i18n",
		"@wordpress/is-shallow-equal": "file:../is-shallow-equal",
		"@wordpress/private-apis": "file:../private-apis",
		"@wordpress/shortcode": "file:../shortcode",
		"change-case": "^4.1.2",
		"colord": "^2.7.0",
		"deepmerge": "^4.3.0",
		"fast-deep-equal": "^3.1.3",
		"hpq": "^1.3.0",
		"is-plain-object": "^5.0.0",
		"memize": "^2.1.0",
		"rememo": "^4.0.2",
		"remove-accents": "^0.4.2",
		"showdown": "^1.9.1",
		"simple-html-tokenizer": "^0.5.7",
		"uuid": "^8.3.0"
	},
	"peerDependencies": {
		"react": "^18.0.0"
	},
	"publishConfig": {
		"access": "public"
	}
}
