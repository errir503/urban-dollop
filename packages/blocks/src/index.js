// A "block" is the abstract term used to describe units of markup that,
// when composed together, form the content or layout of a page.
// The API for blocks is exposed via `wp.blocks`.
//
// Supported blocks are registered by calling `registerBlockType`. Once registered,
// the block is made available as an option to the editor interface.
//
// Blocks are inferred from the HTML source of a post through a parsing mechanism
// and then stored as objects in state, from which it is then rendered for editing.

export { store } from './store';
export * from './api';
export * from './deprecated';
