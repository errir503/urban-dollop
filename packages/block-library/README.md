# Block library

Block library for the WordPress editor.

## Installation

Install the module

```bash
npm install @wordpress/block-library --save
```

_This package assumes that your code will run in an **ES2015+** environment. If you're using an environment that has limited or no support for such language features and APIs, you should include [the polyfill shipped in `@wordpress/babel-preset-default`](https://github.com/WordPress/gutenberg/tree/HEAD/packages/babel-preset-default#polyfill) in your code._

## API

<!-- START TOKEN(Autogenerated API docs) -->

### registerCoreBlocks

Function to register core blocks provided by the block editor.

_Usage_

```js
import { registerCoreBlocks } from '@wordpress/block-library';

registerCoreBlocks();
```

_Parameters_

-   _blocks_ `Array`: An optional array of the core blocks being registered.

<!-- END TOKEN(Autogenerated API docs) -->

## Registering individual blocks

1.  When you only care about registering the block when file gets imported:

    ```js
    import '@wordpress/block-library/build-module/verse/init';
    ```

2.  When you want to use the reference to the block after it gets automatically registered:

    ```js
    import verseBlock from '@wordpress/block-library/build-module/verse/init';
    ```

3.  When you need a full control over when the block gets registered:

    ```js
    import { init } from '@wordpress/block-library/build-module/verse';

    const verseBlock = init();
    ```

## Contributing to this package

This is an individual package that's part of the Gutenberg project. The project is organized as a monorepo. It's made up of multiple self-contained software packages, each with a specific purpose. The packages in this monorepo are published to [npm](https://www.npmjs.com/) and used by [WordPress](https://make.wordpress.org/core/) as well as other software projects.

To find out more about contributing to this package or Gutenberg as a whole, please read the project's main [contributor guide](https://github.com/WordPress/gutenberg/tree/HEAD/CONTRIBUTING.md).

### Adding new blocks

⚠️ Adding new blocks to this package **requires** additional steps!

1.  Do not forget to register a new core block in the [`index.js`](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/index.js) file of this package. For example, if you were to add the new core block called `core/blinking-paragraph`, you would have to add something like:

    ```js
    // packages/block-library/src/index.js
    import * as blinkingParagraph from './blinking-paragraph';
    ```

    Then add `blinkingParagraph` to the list in the `getAllBlocks()` function.

    If it's experimental, add the following property to `block.json`:

    ```json
    {
    	"__experimental": "true"
    }
    ```

2.  Register the block in the `gutenberg_reregister_core_block_types()` function of the [`lib/blocks.php`](https://github.com/WordPress/gutenberg/blob/trunk/lib/blocks.php) file. Add it to the `block_folders` array if it's a [static block](https://developer.wordpress.org/block-editor/explanations/glossary/#static-block) or to the `block_names` array if it's a [dynamic block](https://developer.wordpress.org/block-editor/explanations/glossary/#dynamic-block).

3.  Add `init.js` file to the directory of the new block:

    ```js
    /**
     * Internal dependencies
     */
    import { init } from './';

    export default init();
    ```

    This file is used when using the option to register individual block from the `@wordpress/block-library` package.

4.  If a `view.js` file (or a file prefixed with `view`, e.g. `view-example.js`) is present in the block's directory, this file will be built along other assets, making it available to load from the browser. You only need to reference a `view.min.js` (notice the different file extension) file in the `block.json` file as follows:

    ```json
    {
    	"viewScript": "file:./view.min.js"
    }
    ```

    This file will get automatically loaded when the static block is present on the front end. For dynamic block, you need to manually enqueue the view script in `render_callback` of the block, example:

    ```php
    function render_block_core_blinking_paragraph( $attributes, $content ) {
        $should_load_view_script = ! empty( $attributes['isInteractive'] ) && ! wp_script_is( 'wp-block-blinking-paragraph-view' );
        if ( $should_load_view_script ) {
    	    wp_enqueue_script( 'wp-block-blinking-paragraph-view' );
        }

    	return $content;
    }
    ```

### Naming convention for PHP functions

All PHP function names declared within the subdirectories of the `packages/block-library/src/` directory should start with one of the following prefixes:

-   `block_core_<directory_name>`
-   `render_block_core_<directory_name>`
-   `register_block_core_<directory_name>`

In this context, `<directory_name>` represents the name of the directory where the corresponding `.php` file is located.
The directory name is converted to lowercase, and any characters except for letters and digits are replaced with underscores.

#### Example:

For the PHP functions declared in the `packages/block-library/src/my-block/index.php` file, the correct prefixes would be:

-   `block_core_my_block`
-   `render_block_core_my_block`
-   `register_block_core_my_block`

<br /><br /><p align="center"><img src="https://s.w.org/style/images/codeispoetry.png?1" alt="Code is Poetry." /></p>
