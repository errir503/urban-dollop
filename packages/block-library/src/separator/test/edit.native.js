/**
 * External dependencies
 */
import {
	addBlock,
	getEditorHtml,
	initializeEditor,
	getBlock,
} from 'test/helpers';

/**
 * WordPress dependencies
 */
import { getBlockTypes, unregisterBlockType } from '@wordpress/blocks';
import { registerCoreBlocks } from '@wordpress/block-library';

beforeAll( () => {
	// Register all core blocks
	registerCoreBlocks();
} );

afterAll( () => {
	// Clean up registered blocks
	getBlockTypes().forEach( ( block ) => {
		unregisterBlockType( block.name );
	} );
} );

describe( 'Separator block', () => {
	it( 'inserts block', async () => {
		const screen = await initializeEditor();

		// Add block
		await addBlock( screen, 'Separator' );

		// Get block
		const separatorBlock = await getBlock( screen, 'Separator' );
		expect( separatorBlock ).toBeVisible();
		expect( getEditorHtml() ).toMatchSnapshot();
	} );
} );
