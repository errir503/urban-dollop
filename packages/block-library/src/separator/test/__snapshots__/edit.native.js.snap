// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator block inserts block 1`] = `
"<!-- wp:separator -->
<hr class="wp-block-separator has-alpha-channel-opacity"/>
<!-- /wp:separator -->"
`;
